nodejs/eks/cluster.ts

```diff
@@ -1549,3 +1487,183 @@ export class Cluster extends pulumi.ComponentResource {
         return pulumi.output(kc).apply(JSON.stringify);
     }
 }
+
+/** @internal */
+export interface ClusterResult {
+    kubeconfig: pulumi.Output<any>;
+    awsProvider?: pulumi.ProviderResource;
+    provider: k8s.Provider;
+    clusterSecurityGroup: aws.ec2.SecurityGroup;
+    instanceRoles: pulumi.Output<aws.iam.Role[]>;
+    nodeSecurityGroup: aws.ec2.SecurityGroup;
+    eksClusterIngressRule: aws.ec2.SecurityGroupRule;
+    defaultNodeGroup: NodeGroupData | undefined;
+    eksCluster: aws.eks.Cluster;
+    core: CoreData;
+}
+
+/** @internal */
+export function createCluster(
```

Comment:
This is just copied from the original `Cluster` constructor, so it can be used by both `Cluster` and `ClusterInternal`.